src/cluster.cpp

```diff
@@ -421,6 +424,24 @@ CassError cass_cluster_set_use_randomized_contact_points(CassCluster* cluster,
   return CASS_OK;
 }
 
+CassError cass_cluster_set_constant_speculative_execution_policy(CassCluster* cluster,
+                                                                 cass_int64_t constant_delay_ms,
+                                                                 int max_speculative_executions) {
+  if (constant_delay_ms < 0 || max_speculative_executions < 0) {
+    return CASS_ERROR_LIB_BAD_PARAMS;
+  }
+  cluster->config().set_speculative_execution_policy(
+        new cass::ConstantSpeculativeExecutionPolicy(constant_delay_ms,
+                                                     max_speculative_executions));
+  return CASS_OK;
+}
+
+CassError cass_cluster_set_no_speculative_execution_policy(CassCluster* cluster) {
+  cluster->config().set_speculative_execution_policy(
+        new cass::NoSpeculativeExecutionPolicy());
```

Comment:
A potentially better optimization is to keep returning the same instance of `NoSpeculativeExecutionPlan` (vs policy) since it is allocated for ever request. I've attempted to do this (using custom `operator new` and `operator delete` without luck because multiple threads attempt to write the vtable simultaneously. That tiny allocation doesn't add very much overhead because is likely cached in the allocator anyway.